lib/thrift/generator/struct_binary_protocol.ex

```diff
@@ -912,23 +901,22 @@ defmodule Thrift.Generator.StructBinaryProtocol do
     |> value_serializer(var, file_group)
   end
 
-
-  defp type_id(:bool, _file_group), do: 2
-  defp type_id(:byte, _file_group), do: 3
-  defp type_id(:i8, _file_group), do: 3
-  defp type_id(:double, _file_group), do: 4
-  defp type_id(:i16, _file_group), do: 6
-  defp type_id(:i32, _file_group), do: 8
-  defp type_id(%TEnum{}, _file_group), do: 8
-  defp type_id(:i64, _file_group), do: 10
-  defp type_id(:string, _file_group), do: 11
-  defp type_id(:binary, _file_group), do: 11
-  defp type_id(%Struct{}, _file_group), do: 12
-  defp type_id(%Exception{}, _file_group), do: 12
-  defp type_id(%Union{}, _file_group), do: 12
-  defp type_id({:map, _}, _file_group), do: 13
-  defp type_id({:set, _}, _file_group), do: 14
-  defp type_id({:list, _}, _file_group), do: 15
+  defp type_id(:bool, _file_group),         do: Type.bool
+  defp type_id(:byte, _file_group),         do: Type.byte
+  defp type_id(:i8, _file_group),           do: Type.i8
+  defp type_id(:i16, _file_group),          do: Type.i16
+  defp type_id(:i32, _file_group),          do: Type.i32
+  defp type_id(:i64, _file_group),          do: Type.i64
+  defp type_id(:double, _file_group),       do: Type.double
+  defp type_id(:string, _file_group),       do: Type.string
+  defp type_id(:binary, _file_group),       do: Type.string
+  defp type_id({:map, _}, _file_group),     do: Type.map
+  defp type_id({:set, _}, _file_group),     do: Type.set
+  defp type_id({:list, _}, _file_group),    do: Type.list
+  defp type_id(%TEnum{}, _file_group),      do: Type.i32
+  defp type_id(%Struct{}, _file_group),     do: Type.struct
+  defp type_id(%Exception{}, _file_group),  do: Type.struct
+  defp type_id(%Union{}, _file_group),      do: Type.struct
```

Comment:
How about the non-special cases of these get handled as a fall-through function clause?

```
defp type_id(type, _file_group), do: Type.of(type)
```